### PR TITLE
octanify offering form

### PR DESCRIPTION
**Pull request by stopfstedt**

### CODE REVIEW

File: addon/components/offering-form.js [hunk 1]
```diff
@@ -129,23 +67,25 @@ export default Component.extend(ValidationErrorDisplay, Validations, {
...
+    return mEnd.diff(mStart, 'hours');
+  }
 
-    return diffInHours;
-  }),
-  durationMinutes: computed('startDate', 'endDate', function(){
-    const startDate = this.get('startDate');
-    const endDate = this.get('endDate');
+  @IsInt() @Gte(0) @Lte(59)
+  get durationMinutes() {
```
Comment (jrjohnson) [inline addon/components/offering-form.js]: This is fun! Sort of seems like maybe the future.

File: addon/decorators/validation/gt.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      propertyName: propertyName,
+      constraints: [gt],
+      options: validationOptions,
+      validator: {
+        validate(value, { constraints, property }) {
+          if (!constraints || constraints.length < 1) {
+            throw new Error(`You must pass a ${property} value as the first argument to Gte`);
+          }
+          const numValue = Number(value);
+          if (typeof numValue !== 'number' || isNaN(numValue)) {
```
Comment (jrjohnson) [inline addon/decorators/validation/gt.js]: It doesn't look like `Number()` can ever return anything but a number or `NaN` so the `typeof` comparison seems superfluous.

File: addon/decorators/validation/is-int.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      target: object.constructor,
+      propertyName: propertyName,
+      options: validationOptions,
+      validator: {
+        validate(value) {
+          const numValue = Number(value);
+          if (Number.isInteger) {
+            return Number.isInteger(numValue);
+          }
+          // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger#Polyfill
```
Comment (jrjohnson) [inline addon/decorators/validation/is-int.js]: It looks like this exists in all of our supported browsers, is there a specific one that I missing that would need a polyfill for this?

File: addon/decorators/validation/gt.js (hunk 2 above)
Comment (stopfstedt) [reply addon/decorators/validation/gt.js]: you're right, good catch. yanked from all existing number comparison validators.

File: addon/decorators/validation/is-int.js (hunk 3 above)
Comment (stopfstedt) [reply addon/decorators/validation/is-int.js]: internet exploder doesn't support this method. but even tho we don't support IE anymore, it costs us virtually nothing to polyfill this method. not feeling super strong about this, so if you want this gone then i'll yank it.

File: addon/components/offering-form.js (hunk 1 above)
Comment (stopfstedt) [reply addon/components/offering-form.js]: yea, chaining them is the way to go.

File: addon/decorators/validation/is-int.js (hunk 3 above)
Comment (jrjohnson) [reply addon/decorators/validation/is-int.js]: It's fine with me, you're right no big deal.

**Comment by jrjohnson** [comment]
@dartajax this needs lots of click testing in all the places where an offering can be added or modified. If it passes, it's good to merge.

**Pull request merged**